### ISSUE TITLE
import-aliases asks for overwrite confirmation for every alias being imported

### ISSUE THREAD

**Issue by olets** [olets/zsh-abbr#149]
_Originally posted by @k3n in https://github.com/olets/zsh-abbr/issues/46#issuecomment-2386372852_

In case anyone else finds themself here [olets' edit: #46], when invoking `abbr import-aliases` (which I only found in this issue) it will repeat the following for seemingly _every_ defined alias:

```sh
overwrite /Users/$USER/.config/zsh-abbr/user-abbreviations? (y/n [n])
```

The confusing part is that you'd expect it to ask once, but after several times of answering -- either `y` or `n`, doesn't matter -- you get the feeling like you're in an infinite loop. So you `ctrl-c` and rerun the command, only to face the same response.

On mine, it was even timing out because I have so many (due to OMZ).

One way to handle this is to automatically answer `y` to every prompt:

```sh
yes | abbr import-aliases
```

**Comment by olets** [comment]
> `overwrite /Users/$USER/.config/zsh-abbr/user-abbreviations? (y/n [n])`

@k3n interesting, that's coming from outside of zsh-abbr.

Are you calling `import-aliases` interactively (running `abbr import-aliases` on the command line) or in .zshrc / some other run-every-time zsh setup file?

Does the same thing happen when you use `add` on the command line (e.g. if you run `abbr a=b`)?

> or seemingly every defined alias

I don't expect the confirmation question, but given that you're getting the confirmation question, yes getting it for every alias is expected. `import-aliases` loops over all aliases.

**Comment by k3n** [comment]
Hey, thanks for opening this! I hope it helps others. I'll start from the beginning so we're on the same page.

I learned about this project from reddit, and quickly found your [installation guide](https://zsh-abbr.olets.dev/installation.html) from the project README. However, from experience I know that OMZ usually requires a different setup process for these types of utilities, and I was unable to find any reference to OMZ from the project README or the installation guide so I searched and found [this discussion](https://github.com/olets/zsh-abbr/discussions/105), which led me to the [plugin install](https://gist.github.com/olets/06009589d7887617e061481e22cf5a4a) gist, so I followed that but the expansions weren't working. (even after starting a new shell). So I searched again and found #46, which led me to try calling `abbr import-aliases` directly (and getting the reported result of repeating overwrite confirmations).

I probably did something backwards to make it harder than it should be...

**Comment by k3n** [comment]
> Does the same thing happen when you use `add` on the command line (e.g. if you run `abbr a=b`)?

Not sure I follow exactly, but here's my output:

```sh
❯ abbr a=b
overwrite /Users/$USER/.config/zsh-abbr/user-abbreviations? (y/n [n]) n
not overwritten
Added the regular user abbreviation `a`

❯ abbr aaa=bbb
overwrite /Users/$USER/.config/zsh-abbr/user-abbreviations? (y/n [n]) y
Added the regular user abbreviation `aaa`
```

**Comment by olets** [comment]
Thanks. That confirmation prompt is unexpected in `add` too.

What OS are you on?

What's the output of `abbr profile`?

What installation method did you use? (e.g. plugin manager, Homebrew, manual) Details may be helpful.

Do you encounter similar confirmation prompts in non-zsh-abbr contexts? Is your day filled with being asked whether it's okay to overwrite a file?

What's the output of `whence -v mv`?

**Comment by k3n** [comment]
> What OS are you on?

MacOS 14.7

> What's the output of `abbr profile`?

```
zsh-abbr version 5.8.2
zsh 5.9 (arm-apple-darwin23.0.0)
OSTYPE darwin23.0
```
 
> What installation method did you use? (e.g. plugin manager, Homebrew, manual) Details may be helpful.

Please see [this comment](https://github.com/olets/zsh-abbr/issues/149#issuecomment-2394438493).

> Do you encounter similar confirmation prompts in non-zsh-abbr contexts? Is your day filled with being asked whether it's okay to overwrite a file?

Hah, no :) Never seen this prompt before that I can recall.
 
> What's the output of `whence -v mv`?

```
mv is an alias for mv -i
```

**Comment by olets** [comment]
> Please see https://github.com/olets/zsh-abbr/issues/149#issuecomment-2394438493.

Thanks missed that first comment

> didn't realize that plugin == omz

Reasonable. I've updated the docs

- https://github.com/olets/zsh-abbr-v5-docs/commit/d9f544b76306db8b3cc0eb7aa842e21726982850
- https://zsh-abbr.olets.dev/installation.html#plugin

> mv is an alias for mv -i

I bet that's it. `-i` is "show a confirmation prompt before overwriting a file." Not entirely sure though, because when I add that alias I don't get the confirmation prompt.

I've pushed up a fix attempt. Would you test it for me? (Might need to `cd` to something different in the first command, depending on what exactly you did to install zsh-abbr)

```
cd ${ZSH_CUSTOM:-~/.oh-my-zsh/custom}/plugins/zsh-abbr
git fetch
git checkout issues/149
exec zsh
abbr testing149=x
abbr erase testing149
git checkout -
exec zsh
```

Let me know if you get a confirmation prompts after `abbr testing149=x` or after `abbr erase testing149`. (`git checkout -` and the second `exec zsh` reset things to where they were before testing.)

**Comment by k3n** [comment]
Looks like that addresses it -- I did't get the prompt at all that time! Nice job.

**Comment by k3n** [comment]
Also, seeing same behavior with `import-git-aliases` asking for each abbr -- unsure if proposed fix would apply there as well.

**Comment by olets** [comment]
> seeing same behavior 

Are you saying this did or did not fix `import-git-aliases`?

**Comment by k3n** [comment]
> > seeing same behavior
> 
> Are you saying this did or did not fix `import-git-aliases`?

Oh, no, sorry. I just meant that it had the original problem, I didn't check it against the fix.

**Comment by olets** [comment]
Ok cool. I'm confident that both shared a single cause. Fix is released in [v5.8.3](https://github.com/olets/zsh-abbr/releases/tag/v5.8.3)!

Lmk if you'd like to be listed as a bug report and user testing contributor in the readme and on the docs site's community page.

**Comment by k3n** [comment]
Don't worry about including me -- thanks for the nifty project! Cheers.
